src/Sulu/Component/Content/Document/Subscriber/WebspaceSubscriber.php

```diff
@@ -49,13 +62,39 @@ public function supports($document)
     }
 
     /**
-     * @param HydrateEvent $event
+     * @param AbstractMappingEvent|HydrateEvent $event
+     * @throws \Sulu\Component\DocumentManager\Exception\DocumentManagerException
      */
     public function doHydrate(AbstractMappingEvent $event)
     {
         $document = $event->getDocument();
-        $webspace = $this->inspector->getWebspace($document);
-        $event->getAccessor()->set('webspaceName', $webspace);
+        $webspaceName = $this->inspector->getWebspace($document);
+        $event->getAccessor()->set('webspaceName', $webspaceName);
+
+        if ($document instanceof StructureBehavior && $document->getStructureType() === null) {
+            $this->setDefaultStructureType($event, $document, $webspaceName);
+        }
```

Comment:
so do we remove this here for the time being? #1272 will be the first thing I work on next.